src/sentry/runner/initializer.py

```diff
@@ -230,6 +242,19 @@ def apply_legacy_settings(settings):
                       "Use SENTRY_OPTIONS instead, key 'system.rate-limit'", DeprecationWarning)
         settings.SENTRY_OPTIONS['system.rate-limit'] = settings.SENTRY_SYSTEM_MAX_EVENTS_PER_MINUTE
 
+    if 'redis.clusters' in settings.SENTRY_OPTIONS:
+        if hasattr(settings, 'SENTRY_REDIS_OPTIONS'):
+            raise Exception("Cannot specify both SENTRY_OPTIONS['redis.clusters'] option and SENTRY_REDIS_OPTIONS setting.")
+        # TODO: This probably needs to validate that this configuration is valid somewhere?
```

Comment:
> There's no great way to validate this contains the default key without introducing a new option type.

I wouldn't oppose a `RedisCluster` type that extended `Dict`. (not essential since we don't validate the shape of this data today, so can be done later)

> This seems like (maybe?) the right point in the application to validate each cluster's options (rather than at Cluster initialization, which happens lazily) and there's also no reasonable way to validate that the options are valid arguments for rb.Cluster at this point without actually constructing them without inspect hacks or a option whitelist that needs to be kept in sync with library changes (yuck.)

You referring to actually pinging out to the cluster and making sure they're live? I think that's ok how it's done now.